File: Personal/Roadkill/src/org/roadkill/plugins/intellij/ActivateSelectedEditor.java
package org.roadkill.plugins.intellij;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.editor.Editor;
import com.intellij.openapi.fileEditor.FileEditorManager;
import com.intellij.openapi.wm.IdeFocusManager;

/**
 * User: mlueders
 * Date: 11/30/12 5:26 AM
 */
public class ActivateSelectedEditor extends AnAction {

    public void actionPerformed(AnActionEvent e) {
        Editor editor = FileEditorManager.getInstance(e.getProject()).getSelectedTextEditor();
        if (editor != null) {
            IdeFocusManager focusManager = IdeFocusManager.getInstance(editor.getProject());
            focusManager.requestFocus(editor.getContentComponent(), true);
        }
    }

}

File: Roadkill/src/main/kotlin/org/roadkill/plugins/intellij/ActivateSelectedEditor.kt
package org.roadkill.plugins.intellij

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.fileEditor.FileEditorManager
import com.intellij.openapi.wm.IdeFocusManager


class ActivateSelectedEditor : AnAction() {

    override fun actionPerformed(e: AnActionEvent) {
        val editor = FileEditorManager.getInstance(e.project!!).selectedTextEditor
        if (editor != null) {
            val focusManager = IdeFocusManager.getInstance(editor.project)
            focusManager.requestFocus(editor.contentComponent, true)
        }
    }

}
